rust/relay/src/server.rs

```diff
@@ -377,15 +401,15 @@ where
             return;
         }
 
-        tracing::trace!(target: "wire", num_bytes = %bytes.len());
+        tracing::trace!(target: "wire", num_bytes = %msg.len());
 
-        self.data_relayed_counter.add(bytes.len() as u64, &[]);
-        self.data_relayed += bytes.len() as u64;
+        self.data_relayed_counter.add(msg.len() as u64, &[]);
+        self.data_relayed += msg.len() as u64;
 
-        let data = ChannelData::new(*channel_number, bytes).into_msg();
```

Comment:
This was the 2nd allocation. Now gone because we pre-allocated enough space to write the header into the same allocation.